lib/exhal/client.ex

```diff
@@ -5,57 +5,88 @@ defmodule ExHal.Client do
   ## Examples
 
       iex> ExHal.Client.new()
-      %ExHal.Client{}
+      ...> |> ExHal.Client.get("http://haltalk.herokuapp.com/")
+      %ExHal.Document{...}
+
+      iex> ExHal.Client.new()
+      ...> |> ExHal.Client.post("http://haltalk.herokuapp.com/signup", ~s(
+      ...>      { "username": "fred",
+      ...>        "password": "pwnme",
+      ...>        "real_name": "Fred Wilson" }
+      ...>     ))
+      %ExHal.Document{...}
+
+      iex> authorizer = ExHal.SimpleAuthorizer.new("http://haltalk.herokuapp.com",
+      ...>                                         "Bearer my-token")
+      iex> ExHal.Client.new()
+      ...> |> ExHal.Client.add_headers("Prefer": "minimal")
+      ...> |> ExHal.Client.set_authorizer(authorizer)
+      %ExHal.Client{...}
   """
 
   require Logger
-  alias ExHal.{Document, NonHalResponse, ResponseHeader}
+  alias ExHal.{Document, NonHalResponse, ResponseHeader, Authorizer, NullAuthorizer}
 
   @logger Application.get_env(:exhal, :logger, Logger)
+  @http_client Application.get_env(:exhal, :http_client, HTTPoison)
```

Comment:
Seems like this could have unexpected behavior because attributes are compile time only. I think I usually don't change config between compile- and run-time, but that seems like coincidence.

Reply:
I haven't explored how this works but it is the way Jose suggests doing it, <http://blog.plataformatec.com.br/2015/10/mocks-and-explicit-contracts/>.